packages/nx-plugin/src/generators/add-to-project/generator.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  Tree,
+  formatFiles,
+  generateFiles,
+  readProjectConfiguration,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import * as path from 'path';
+import { join } from 'path';
+import { AddToProjectGeneratorSchema } from './schema';
+
+export async function addToProjectGenerator(
+  tree: Tree,
+  options: AddToProjectGeneratorSchema,
+) {
+  const projectConfiguration = readProjectConfiguration(tree, options.project);
+
+  const { root } = projectConfiguration;
+
+  if (tree.exists(path.join(root, 'code-pushup.config.json'))) {
+    console.log('Code PushUp already configured for this project');
+    return;
+  }
+
+  generateFiles(tree, path.join(__dirname, 'files'), root, options);
+
+  updateProjectConfiguration(tree, options.project, {
+    ...projectConfiguration,
+    targets: {
+      ...projectConfiguration.targets,
+      collect: {
```

Comment:
Would it make sense to name the target code-pushup? As we can run different commands. 
 (`autorun` is default)